frame/staking/src/pallet/impls.rs

```diff
@@ -662,9 +661,12 @@ impl<T: Config> Pallet<T> {
 		slashing::clear_stash_metadata::<T>(stash, num_slashing_spans)?;
 
 		<Bonded<T>>::remove(stash);
-		<Ledger<T>>::remove(&controller);
-
 		<Payee<T>>::remove(stash);
+		Self::ledger(&controller).map_or_else(
```

Comment:
Shouldn't we never use this getter (`Self::ledger`) anymore after this refactor?